File: 2025/01/main_test.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"strings"
	"testing"
)

func Test_Process(t *testing.T) {
	raw := "L68\nL30\nR48\nL5\nR60\nL55\nL1\nL99\nR14\nL82\n"

	reader := strings.NewReader(raw)
	scanner := bufio.NewScanner(reader)
	if scanner.Err() != nil {
		log.Fatalln(fmt.Errorf("scanner error => %w", scanner.Err()))
	}

	var data []string
	for scanner.Scan() {
		line := scanner.Text()
		data = append(data, line)
	}

	firstExpected := int64(3)
	firstActual := first(data)
	if firstExpected != firstActual {
		t.Errorf("expected: %d, actual: %d", firstExpected, firstActual)
	}

	secondExpected := int64(6)
	secondActual := second(data)
	if secondExpected != secondActual {
		t.Errorf("expected: %d, actual: %d", secondExpected, secondActual)
	}
}

File: 2025/04/main_test.c
#include <stddef.h>
#include <stdio.h>

#include "lib.c"

void assert(int expected, int actual) {
    if (expected == actual)
        printf("Assertion Passed.\n");
    else
        printf("Assertion Failed; expected: %d, actual: %d\n", expected, actual);
}

int main() {
    size_t m = 10;
    size_t n = 10;

    const char* data[10] = {
        "..@@.@@@@.", "@@@.@.@.@@", "@@@@@.@.@@", "@.@@@@..@.", "@@.@@@@.@@",
        ".@@@@@@@.@", ".@.@.@.@@@", "@.@@@.@@@@", ".@@@@@@@@.", "@.@.@@@.@.",
    };

    uint64_t first_expected = 13;
    uint64_t first_actual = first(data, n);
    assert(first_expected, first_actual);

    uint64_t second_expected = 43;
    uint64_t second_actual = second(data, n);
    assert(second_expected, second_actual);
}
